src/core/entity/category-ecommerce.entity.ts

```diff
@@ -0,0 +1,19 @@
+import { Column, Entity, ManyToOne, OneToMany, PrimaryGeneratedColumn } from 'typeorm';
+import { EcommerceEntity } from './ecommerce.entity';
+import { ProductEcommerceEntity } from './product-ecommerce.entity';
+
+@Entity('category_ecommerce')
+export class CategoryEcommerceEntity {
+  @PrimaryGeneratedColumn()
+  id!: number;
+
+  @Column()
+  name!: string;
+
+  @ManyToOne(() => EcommerceEntity, (ecommerce) => ecommerce.categories, { onDelete: 'CASCADE' })
+  ecommerce!: EcommerceEntity;
+
+  @OneToMany(() => ProductEcommerceEntity, (product) => product.category, { onDelete: 'CASCADE', eager: true })
```

Comment:
```suggestion
  @OneToMany(() => ProductEcommerceEntity, (product) => product.category, { onDelete: 'CASCADE' })
```
I found that this option can generate problem in time respond of the services